sys/net/application_layer/dhcpv6/client.c

```diff
@@ -817,15 +832,19 @@ static void _request_renew_rebind(uint8_t type)
         case DHCPV6_REBIND: {
             irt = DHCPV6_REB_TIMEOUT;
             mrt = DHCPV6_REB_MAX_RT;
-            /* calculate MRD from prefix leases */
-            for (unsigned i = 0; i < CONFIG_DHCPV6_CLIENT_PFX_LEASE_MAX; i++) {
-                const pfx_lease_t *lease = &pfx_leases[i];
-                uint32_t valid_until = dhcpv6_client_prefix_valid_until(
-                        lease->parent.ia_id.info.netif,
-                        &lease->pfx, lease->pfx_len
-                    );
-                if (valid_until > mrd) {
-                    mrd = valid_until;
+            if (IS_USED(MODULE_DHCPV6_CLIENT_IA_PD)) {
+                /* calculate MRD from prefix leases */
+                for (unsigned i = 0;
+                     i < CONFIG_DHCPV6_CLIENT_PFX_LEASE_MAX;
```

Comment:
You could also do

```suggestion
                for (unsigned i = 0;
                     IS_USED(MODULE_DHCPV6_CLIENT_IA_PD) &&
                     (i < CONFIG_DHCPV6_CLIENT_PFX_LEASE_MAX);
```

as a above and save a level of indentation